File: src/main/java/com/ptaf/ai/validation/YamlKeyValidationResult.java
package com.ptaf.ai.validation;

import java.util.List;
import java.util.Map;

public record YamlKeyValidationResult(
        List<String> yamlKeysUsed,
        List<String> existingKeys,
        List<String> missingKeys,
        Map<String, String> suggestedPatches,
        int totalKeys,
        int existingCount,
        int missingCount,
        boolean passed,
        List<String> warnings
) {
}

File: src/main/ptaf/ai/validation/yaml_key_validation_result.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class YamlKeyValidationResult:
    yaml_keys_used: list[str]
    existing_keys: list[str]
    missing_keys: list[str]
    suggested_patches: dict[str, str]
    total_keys: int
    existing_count: int
    missing_count: int
    passed: bool
    warnings: list[str]
